device/core/src/client.ts

```diff
@@ -292,12 +344,13 @@ export class Client extends EventEmitter {
     /*Codes_SRS_NODE_DEVICE_CLIENT_16_042: [The `setOptions` method shall throw a `ReferenceError` if the options object is falsy.]*/
     if (!options) throw new ReferenceError('options cannot be falsy.');
 
-    this._transport.setOptions(options, (err) => {
+    const retryOp = new RetryOperation(this._retryPolicy, this.maxOperationTimeout);
```

Comment:
>RetryOperation [](start = 24, length = 14)

It's not intuitive to me why setting options would need to be a retryOp.  I assume it's because this might trigger a transport reconnection?  Or is there some other reason?  It might be useful to add a comment.